x-pack/plugins/search_indices/public/hooks/use_elasticsearch_url.ts

```diff
@@ -14,5 +15,13 @@ export const useElasticsearchUrl = (): string => {
     services: { cloud },
   } = useKibana();
 
-  return cloud?.elasticsearchUrl ?? ELASTICSEARCH_URL_PLACEHOLDER;
+  const [elasticsearchUrl, setElasticsearchUrl] = useState<string>(ELASTICSEARCH_URL_PLACEHOLDER);
+
+  useEffect(() => {
+    cloud?.fetchElasticsearchConfig().then((config) => {
+      setElasticsearchUrl(config?.elasticsearchUrl || ELASTICSEARCH_URL_PLACEHOLDER);
+    });
+  }, [cloud]);
+
+  return elasticsearchUrl ?? ELASTICSEARCH_URL_PLACEHOLDER;
```

Comment:
this ` ?? ELASTICSEARCH_URL_PLACEHOLDER;` is redundant now that `elasticsearchUrl` is set from state it will always have a value.